cuda/btx_cudainterval_callbacks.cpp

```diff
@@ -0,0 +1,589 @@
+                              uint64_t vtid, bool err) {
+  std::string event_class_name_striped = strip_event_class_name(event_class_name);
+  const int64_t _start = get_entry_ts(static_cast<data_t *>(usr_data), {hostname, vpid, vtid});
+
+  btx_push_message_lttng_host(btx_handle, hostname, vpid, vtid, _start, BACKEND_CUDA,
+                              event_class_name_striped.c_str(), (ts - _start), err);
+}
+
+static void send_traffic_message(void *btx_handle, void *usr_data, int64_t ts,
+                                 const char *event_class_name, const char *hostname,
+                                 int64_t vpid, uint64_t vtid, bool err) {
+  std::string event_class_name_stripped = strip_event_class_name(event_class_name);
+  auto state = static_cast<data_t *>(usr_data);
+  hpt_t key = {hostname, vpid, vtid};
+  const int64_t _start = get_entry_ts(state, key);
+
+  if (!err) {
+    auto size = state->entry_traffic_size.at(key);
+    btx_push_message_lttng_traffic(btx_handle, hostname, vpid, vtid, _start, BACKEND_CUDA,
+                                   event_class_name_stripped.c_str(), size);
+  }
+}
+
+
+void btx_initialize_component(void **usr_data) { *usr_data = new data_t; }
+
+void btx_finalize_component(void *usr_data) {
+  delete static_cast<data_t *>(usr_data);
+}
+
+void entries_callback(void *btx_handle, void *usr_data, int64_t ts,
+                      const char *event_class_name, const char *hostname, int64_t vpid,
+                             uint64_t vtid) {
+  set_entry_ts(((data_t *)usr_data), {hostname, vpid, vtid}, ts);
+}
+
+void exits_callback_cudaError_absent(void *btx_handle, void *usr_data, int64_t ts,
+                                     const char *event_class_name, const char *hostname,
+                                     int64_t vpid, uint64_t vtid) {
+  hpt_t hpt = {hostname, vpid, vtid};
+  try {
+    send_host_message(btx_handle, usr_data, ts, event_class_name,
+                      hostname, vpid, vtid, false);
+  } catch(const std::out_of_range& oor) {
+    THAPI_FATAL_HPT(hpt,
+      "entry ts not found for thread in exit cudaError_absent callback");
+  }
+}
+
+void exits_callback_cudaError_present(void *btx_handle, void *usr_data, int64_t ts,
+                                      const char *event_class_name, const char *hostname,
+                                      int64_t vpid, uint64_t vtid, CUresult cuResult) {
+  hpt_t hpt = {hostname, vpid, vtid};
+  // Not an Error (cuResult == cudaErrorNotReady)
+  bool err = (cuResult != CUDA_SUCCESS) && (cuResult != CUDA_ERROR_NOT_READY);
+  try {
+    send_host_message(btx_handle, usr_data, ts, event_class_name,
+                      hostname, vpid, vtid, err);
+  } catch(const std::out_of_range& oor) {
+    THAPI_FATAL_HPT(hpt,
+                    "entry ts not found for thread in exit cudaError_present callback");
+  }
+}
+
+void entries_traffic_v2_callback(void *btx_handle, void *usr_data, int64_t ts,
```

Comment:
Please add a  comment to say that the only difference at the type of the size :)